### ISSUE TITLE
Pivot tables should support all databases

### ISSUE THREAD

**Issue by paulrosenzweig** [metabase/metabase#15008]
Pivot tables currently require a database driver that supports both `expressions` and `left-join`. We should relax both of those constraints. Fundamentally, any database that can do aggregations should be able to produce a pivot table.

`expressions` – We require this so we can put the "pivot-grouping" number in the query itself. I think we should be able to add that column to the dataset after we get the results from the database. That would enable to keep the pivot grouping format even on databases that don't support expressions.

`left-join` – I'm not sure why this is required. I think I added the limitation to the frontend based on [this](https://github.com/metabase/metabase/blob/master/test/metabase/api/pivots.clj#L10). If that limitation is just for the backend _tests_ rather than the backend code we can remove the frontend check.

**Comment by marcoruggine** [comment]
Thanks for opening this, Pivots in BigQuery are definitely a game-changer for us!

**Comment by marcoruggine** [comment]
Hi guys, any news on this issue?

**Comment by nitu-group** [comment]
Also in the meantime is there a list of which db connectors are supported?

**Comment by nitu-group** [comment]
> Also in the meantime is there a list of which db connectors are supported?

Discussion about compatible databases seems to have started here: https://github.com/metabase/metabase/issues/14876

**Comment by paoliniluis** [comment]
Related to https://github.com/metabase/metabase/issues/4432

**Comment by GJMcClintock** [comment]
Also looking for BigQuery pivot table help - it sounds like that's not quite feasible yet unfortunately.

Anybody found a workaround?

**Comment by unais** [comment]
Anyone found any workaround for this

**Comment by marcoruggine** [comment]
@unais I've tried this one: https://hoffa.medium.com/easy-pivot-in-bigquery-one-step-5a1f13c6c710
works in BQ but unfortunately it's not a good fit for our needs

**Comment by unais** [comment]
Thanks, Marco for the suggestion, but I don't want to create a new table or table view for the pivot

**Comment by marcoruggine** [comment]
Yep, that's my point as well. Also, we need to be able to map more than just 2 columns.

**Comment by JGrubb** [comment]
Hey team, I don't want to nag, but I check this issue at least twice a week to see if there's been any movement.  Is there any way that we can help with this?

**Comment by marcoruggine** [comment]
Hi Team, any news on this issue? In my company this is causing increasing discontent and it's the major point in discussions for migrating away from Metabase. After so many years I'd hate for this to happen.

**Comment by flamber** [comment]
We're working on updating the BigQuery driver, which means it should get the required functionality for Pivot Table.
I can see it's partially working in development builds, but it still needs some work to make actually function.
It is currently unknown if it will make it for 0.41.

Please upvote the issue by clicking :+1: on the first comment and subscribe to notifications in the top-sidebar of this issue.

**Comment by marcoruggine** [comment]
thanks @flamber for the quick reply, glad to hear you are working on it!

**Comment by flamber** [comment]
The new BigQuery driver in 0.41 will support Pivot Table. There's currently no timeline for other unsupported databases.
![image](https://user-images.githubusercontent.com/1447303/135584139-f6050fc9-bdc1-40fe-8a76-a6f74736282a.png)

**Comment by marcoruggine** [comment]
This is great news for us BQ users! Thanks @flamber

**Comment by satyamdorville** [comment]
And for MongoDb now ?

**Comment by flamber** [comment]
@xuheng925 There is no timeline for Mongo. It would require a major refactor of Pivot Table, or we have to add support for Custom Column etc to Mongo. Big projects no matter what.

**Comment by xuheng925** [comment]
> @xuheng925 There is no timeline for Mongo. It would require a major refactor of Pivot Table, or we have to add support for Custom Column etc to Mongo. Big projects no matter what.

I believe metabase is the best fit for us until reading the pivot table section in the help document where I heartbrokenly found pivot table is not support with mongodb >_<

do you have a better workaround for this?

**Comment by camsaul** [comment]
No idea why `:left-join` should be a required feature to run pivot queries honestly. MongoDB supports `:expressions`, and now that we have the Druid JDBC driver it should as well going forward

**Comment by dragonsahead** [comment]
https://github.com/metabase/metabase/pull/76560 should address this for all mbql5 supported databases
